src/Command/PluginUnloadCommand.php

```diff
@@ -42,55 +50,42 @@ public static function defaultName(): string
      */
     public function execute(Arguments $args, ConsoleIo $io): ?int
     {
-        $plugin = $args->getArgument('plugin');
-        if (!$plugin) {
-            $io->err('You must provide a plugin name in CamelCase format.');
-            $io->err('To unload an "Example" plugin, run `cake plugin unload Example`.');
-
-            return static::CODE_ERROR;
-        }
-
-        $app = APP . 'Application.php';
-        if (file_exists($app) && $this->modifyApplication($app, $plugin)) {
-            $io->out('');
-            $io->out(sprintf('%s modified', $app));
+        $plugin = (string)$args->getArgument('plugin');
 
+        $result = $this->modifyConfigFile($plugin);
+        if ($result === null) {
             return static::CODE_SUCCESS;
         }
 
+        $io->err($result);
+
         return static::CODE_ERROR;
     }
 
     /**
-     * Modify the application class.
+     * Modify the plugins config file.
      *
-     * @param string $app Path to the application to update.
-     * @param string $plugin Name of plugin.
-     * @return bool If modify passed.
+     * @param string $plugin Plugin name.
+     * @return string|null
      */
-    protected function modifyApplication(string $app, string $plugin): bool
+    protected function modifyConfigFile(string $plugin): ?string
     {
-        $plugin = preg_quote($plugin, '/');
-        $finder = "/
-            # whitespace and addPlugin call
-            \s*\\\$this\-\>addPlugin\(
-            # plugin name in quotes of any kind
-            \s*['\"]{$plugin}['\"]
-            # method arguments assuming a literal array with multiline args
-            (\s*,[\s\\n]*\[(\\n.*|.*){0,5}\][\\n\s]*)?
-            # closing paren of method
-            \);/mx";
+        // phpcs:ignore
+        $config = @include $this->configFile;
```

Comment:
I think it would be helpful to throw an error on missing file for unload. If they haven't converted, users won't understand the error  below.

Reply:
Done